### PR TITLE
Upstreaming  11 - The rest

### PR DESCRIPTION
**Pull request by CKolkey**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
```
Comment (ten3roberts) [inline .editorconfig:2]: Which plugin do you use for indent detection?

I've seen both modeline, editorconfig, vim-sleuth, as well as recent lua alternatives.

File: todo.md [hunk 2]
```diff
@@ -22,11 +21,10 @@
 ## Popups
 
 * chore: rename popup -> transient
-* fix: default values for options don't get displayed on initial draw
-
-## Stash
-
-* feat: delete stash
+* refactor: action/switch/option/config should all exist on the same list within state, allowing better control over
```
Comment (ten3roberts) [inline todo.md:24]: Absolutely love this. Having a "list" rather than an implementation makes it much easier to a see and audit

**Review by ten3roberts — APPROVED** [review]
Note: formatting and `<esc>` to hide process console

File: .editorconfig (hunk 1 above)
Comment (CKolkey) [reply .editorconfig:2]: Just using the (now) build-in editorconfig support + https://github.com/nmac427/guess-indent.nvim

File: todo.md (hunk 2 above)
Comment (CKolkey) [reply todo.md:24]: Agreed. It's _slightly_ hacky right now with the different columns/rows and headers. Works fine, but I totally agree, a list of components is much simpler and lets you do more.

Currently, something I want to do (and can't) is to have _multiple_ 2D-grids within a popup. The current implementation limits you to the actions group at the bottom, and this works well enough that I would consider it a very low-priority fix, but still.. can't do it.

**Comment by CKolkey** [comment]
Alright, with this merged to CKolkey, I'll:

- backport all the changes to my fork
- open one last PR with any remaining changes _not yet on CKolkey_ (I did do some more stuff before I made a development branch on my fork)
- Open final PR with the full commit history

**Pull request merged**